### PR TITLE
Remove iOS 13 Widgets

### PR DESCRIPTION
**Pull request by mokagio**
Removes the iOS 13 widgets, WordPressTodayWidget, WordPressThisWeekWidget, and WordPressAllTimeWidget, plush does as little work as possible to get the app to build with iOS 14 as the minimum deployment target (#19496).

See also commit messages for additional details on the rationale behind some of the changes.

While working on this, I addressed a few more deprecations, see #19512, but I decided to keep them separated to focus this PR on widgets.

## Regression Notes

1. Potential unintended areas of impact – N.A.
2. What I did to test those areas of impact (or what existing automated tests I relied on) – N.A.
3. What automated tests I added (or what prevented me from doing so) – N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

### CODE REVIEW

File: WordPress/Info.plist
Comment (mokagio) [inline WordPress/Info.plist:554]: As far as I could gather from [the docs](https://developer.apple.com/documentation/bundleresources/information_property_list/uiapplicationshortcutwidget) and the latest [support page](https://support.apple.com/en-us/HT207122), this key was used at the time of the iOS 10 to 13 widgets but is no longer needed.

File: WordPress/WordPress.xcodeproj/project.pbxproj
Comment (mokagio) [inline WordPress/WordPress.xcodeproj/project.pbxproj]: Fun fact. We've been carrying around these `WordPressHomeWidgetToday` dead references ever since renaming that target to `WordPressTodayWidget` 😳

Comment (staskus) [reply WordPress/WordPress.xcodeproj/project.pbxproj]: Thanks for noticing. The project will get a well-deserved clean-up then  👀

**Comment by mokagio** [comment]
cc @crazytonyli, if you have the time to take a look.

**Review by staskus — COMMENTED** [review]
Thank you for the changes, @mokagio! I reviewed commit by commit. Additional changes unrelated to Widgets make sense as well. 👍
 
⚠️ Additional files that can be removed (In Today Widgets / Shared Views):
- `WidgetDiffferenceCell`
- `WidgetTwoCollumnCell`
- `WidgetUnconfiguredCell`
- `WidgetUrlCell`
- `WidgetNoConnectionCell`

Potentially, there could be minor details related to iOS13 Widgets that could be removed as well, for example, some Constants and details saved to UserDefaults. However, iOS13 Widget data-loading code is reused by iOS14+ Widgets, therefore, it's hard to tell if any small details could be removed without negatively affecting current widgets. The majority of the code is removed in this PR. 

⚠️ I see there're multiple places that still have `#available(iOS 14.0, *)`, including Stats-related `StatsWidgetStore`. However, they could be removed in the parent PR or later. 

I tested the app:
- iOS 14 widgets continue working
- Existing shortcuts continue working

**Comment by mokagio** [comment]
Thank you for the review @staskus 🙌 

I removed the Shared Views that this work made unused as you suggested.

> I see there're multiple places that still have `#available(iOS 14.0, *)`, including Stats-related `StatsWidgetStore`. However, they could be removed in the parent PR or later.

Agreed. I might leave them for a PR later on. As a matter of fact, I already removed them [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/19512/commits/3069cb7fd81be9d130ddf444c58208e15c3b50d1) but I think I'll submit them in isolation after #19496 lands.

**Review by staskus — COMMENTED** [review]
@mokagio, [I don't see SharedViews being removed](https://github.com/wordpress-mobile/WordPress-iOS/tree/mokagio/remove-ios-13-widgets/WordPress/Classes/ViewRelated/Stats/Today%20Widgets/Shared%20Views).

**Review by staskus — APPROVED** [review]
I don't want to block this PR, so after shared views are removed, we can move on. 👍

**Comment by mokagio** [comment]
> @mokagio, [I don't see SharedViews being removed](https://github.com/wordpress-mobile/WordPress-iOS/tree/mokagio/remove-ios-13-widgets/WordPress/Classes/ViewRelated/Stats/Today%20Widgets/Shared%20Views?rgh-link-date=2022-10-27T06%3A41%3A31Z).

I didn't push them... 😓 But you can see them in the timeline now

image

Thank you for the approval to unblock. I'll merge as soon as CI is green.

**Comment by staskus** [comment]
I see the changes now, all good 🚀

**Review by guarani — APPROVED** [review]
Apologies for the late review, @mokagio. The code changes look right and I gave the Jetpack app and the WP app both a quick smoke test (adding Home Screen widgets, scheduling a post, etc.) – all good.

I love how this reduced the number of targets in the project ❤️  Thanks for wrangling this!

Are you planning to merge this in 21.1 or bump to the next version?

**Pull request merged**
